DirectProgramming/DPC++/StructuredGrids/1d_HeatTransfer/src/1d_HeatTransfer.cpp

```diff
@@ -60,114 +60,177 @@ void Usage(string programName) {
   cout << " i : Number of timesteps \n";
 }
 
-//************************************
-// Function description: initialize the array.
-//************************************
-void Initialize(float* array, unsigned int num, unsigned int idx) {
-  for (unsigned int i = idx; i < num; i++) array[i] = 0.0f;
+//
+// Initialize the temperature arrays
+//
+void Initialize(float* arr, float* arr_next, unsigned int num) {
+  arr[0] = arr_next[0] = initial_temperature;
+  for (unsigned int i = 0; i < num; i++)
```

Comment:
Should be i=1 to not overwrite initial_temperature.

Reply:
Fixed